vm/systemSmartContracts/stakingWaitingList.go

```diff
@@ -835,11 +838,42 @@ func (s *stakingSC) unStakeAllNodesFromQueue(args *vmcommon.ContractCallInput) v
 		// delete element from waiting list
 		inWaitingListKey := createWaitingListKey(blsKey)
 		s.eei.SetStorage(inWaitingListKey, nil)
+
+		ownerAddr := string(registrationData.OwnerAddress)
+		_, exists := mapOwnerKeys[ownerAddr]
+		if !exists {
+			mapOwnerKeys[ownerAddr] = make([][]byte, 0)
+			orderedListOwners = append(orderedListOwners, ownerAddr)
+		}
+
+		mapOwnerKeys[ownerAddr] = append(mapOwnerKeys[ownerAddr], blsKey)
 	}
 
 	// delete waiting list head element
 	s.eei.SetStorage([]byte(waitingListHeadKey), nil)
 
+	// call unStakeAtEndOfEpoch from the delegation contracts to compute the new unStaked list
+	for _, owner := range orderedListOwners {
```

Comment:
[question]
Why is it mandatory to unstake them ordered?

Reply:
In order to have a deterministic execution.